extensions/amp-auto-ads/0.1/ad-network-config.js

```diff
@@ -65,16 +67,16 @@ class AdSenseNetworkConfig {
 
   /** @override */
   getConfigUrl() {
+    const docInfo = documentInfoForDoc(this.autoAmpAdsElement_);
+    const canonicalHostname = parseUrl(docInfo.canonicalUrl).hostname;
```

Comment:
Canonical URL is provided by publisher. It's possible multiple AMP docs pointing to the same canonical URL. Will that cause a problem for you?